src/state/reducer.ts

```diff
@@ -4,7 +4,7 @@ import type { ArweaveTransactionId, ArNSContractState } from '../types';
 export type Action =
   | { type: 'setWalletAddress'; payload: ArweaveTransactionId }
   | { type: 'setGateway'; payload: string }
-  | { type: 'setArnsContractState'; payload: ArNSContractState };
+  | { type: 'setArnsContractState'; payload: ArNSContractState | unknown };
```

Comment:
why is this needed? f it's complaining about the type, we need to make sure we're only dispatching a valid contract state
```suggestion
  | { type: 'setArnsContractState'; payload: ArNSContractState };
```

Reply:
i took this out too